### ISSUE TITLE
Possible leakage of Vorto user e-mail addresses

### ISSUE THREAD

**Issue by mena-bosch** [eclipse/vorto#2566]
There is an old potential leak of Vorto users' e-mail addresses through the `/rest/accounts/{username:.+}` GET endpoint in the `AccountController`, that has never been intercepted. 

The endpoint only requires authentication as pre-authorization, and with any valid username, will return a payload consisting in the username, but also some more sensitive data such as e-mail address or name of the authentication provider. 

The endpoint is used in many front-end controllers, but it seems the e-mail address (by far, the _most_ sensitive bit of information) is only retrieved in the account settings, i.e. to display the e-mail address of the logged-on user, if they have set it themselves. 

The second sensitive bit of information - the authentication provider - is displayed in the namespace request access form when parametrized with a user's name. While this is in fact technically a leakage (and a new one, compared to the rest), it has very little value as is, and might not be resolvable. 

While this leak has been there forever, it is slightly aggravated by the recent implementation of search combo-boxes for users, which could potentially be employed to brute-force a list of Vorto usernames out for any authenticated user bothering to do so.

----  

I think a good compromise would be to suppress any sensitive data from the payload returned by GET `/rest/accounts/{username:.+}`, except when the authenticated username matches the `{username:.+}` path variable. 
That should not break other usages of the endpoint. 

Minor leakages, include retrieving namespace collaborators' authentication provider (but not e-mail) when managing the collaborators for a given namespace. As they are queriable by REST and pre-authorize with any authenticated user, they could also be subject to an additional authorization check to verify the calling user has the `namespace_admin` role on the given namespace. 

# TL;DR

- Suppress e-mail and authentication provider from user retrieved through GET `/rest/accounts/{username:.+}` if `username` path variable does not match calling user's name
- Make retrieval of namespace collaborators with GET `/rest/namespaces/{namespace:.+}/users`available only to calling users who have the `namespace_admin` role on the namespace.

@t-gauss @kolotu thoughts?

**Comment by mena-bosch** [comment]
Note on 

> Make retrieval of namespace collaborators with GET /rest/namespaces/{namespace:.+}/users available only to calling users who have the namespace_admin role on the namespace.

I forgot this was already suppressing user e-mails by design. While it still shows the username + authentication provider properties for all users of any given namespace, and is accessible to any authenticated user, it does not expose an unreasonable amount of data in my opinion, so it can be probably left as is, or fine-tuned if deemed necessary in another task with lower priority.
